source/administration/config-settings.rst

```diff
@@ -2093,6 +2093,15 @@ See the `Guest Accounts documentation <https://docs.mattermost.com/deployment/gu
 | This feature's ``config.json`` setting is ``"GuestFilter": ""`` with string input.                                                                                   |
 +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 
+Admin Filter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+*Available in Enterprise Edition E20*
+
+(Optional) Enter a filter to use for designating the System Admin role to users. The user is promoted to this role on their next login or at the next scheduled AD/LDAP sync.
```

Comment:
What happens to users who were assigned an admin role and do not match the filter?

Reply:
Good question. Let me know what you think of this: 

"If Admin access is set to false, users who are currently logged in retain their Admin role. When they log out this is revoked and on their next login they will no longer have Admin privileges."

Reply:
Adding @thefactremains and @sbishel to this for their input. :)

Reply:
@thefactremains - This change to two fields. A boolean to "enable" the functionality and a string to store the filter itself. 

"EnableAdminAttribute": false,
"AdminAttribute": "",

According to test steps. The role should remain if the enable button is unchecked.
https://mattermost.atlassian.net/browse/MM-20061

Reply:
Thanks @sbishel.

So based on the test steps, the user retains the admin role if the enable button is unchecked. But if they are currently an admin and no filter is set, then their role is downgraded. 

In this case, I think this might work better: "If the admin filter is removed, users who are currently logged in retain their Admin role. When they log out this is revoked and on their next login they will no longer have Admin privileges." What do you think, @wiersgallak ?

Reply:
@sbishel, is this correct:
SAML: "This feature's ``config.json`` setting is ``"EnableAdminAttribute": false"`` with string input."
LDAP: "This feature's ``config.json`` setting is ``"EnableAdminFilter": false"`` with string input."

Reply:
Forgot to add that "string input would be "true" or "false"" - but not sure if I am understanding the boolean/string combination.

Reply:
@justinegeffen  this works for me. 

> "If the admin filter is removed, users who are currently logged in retain their Admin role. When they log out this is revoked and on their next login they will no longer have Admin privileges."